extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/cors/CORSFilter.java

```diff
@@ -154,8 +154,7 @@ public void handle(RoutingContext event) {
                 response.setStatusCode(403);
                 response.setStatusMessage("CORS Rejected - Invalid origin");
             } else {
-                boolean allowCredentials = corsConfig.accessControlAllowCredentials
-                        .orElse(corsConfig.origins.isPresent() && corsConfig.origins.get().contains(origin));
+                boolean allowCredentials = corsConfig.accessControlAllowCredentials.orElse(originMatches);
```

Comment:
I'm revisiting the docs, I think the original intention was to enable this rather sensitive response header only if a precise match on the Origin is done. If we allow the regex then it can be as wide as the configured wildcard, in which case only checking against `allowsOrigin` is sufficient.

Reply:
Hmmm, that's the reason why I did it that way. I think the regexp and * are rather different beasts and for now I was intentionally opening the gate only for the regexp (which might be used when you have several subdomains for instance).

Reply:
@gsmet What I was thinking, was that if an origin like the one suggested at https://quarkus.io/guides/security-cors#support-all-origins-in-dev-mode is retained in prod, it is a regex, but it is no different to a non-regex `*`. 

Can we try to exclude a possibility of allowing credentials if `/.*/` was used for the successful origin match check. It won't prevent users from doing other very loose expressions, but at least we will minimize a chance of users opening it up unintentionally...